crates/polars-core/src/chunked_array/logical/categorical.rs

```diff
@@ -191,15 +192,76 @@ impl<T: PolarsCategoricalType> CategoricalChunked<T> {
     }
 
     pub fn to_arrow(&self, compat_level: CompatLevel) -> DictionaryArray<T::Native> {
-        let keys = self.physical().rechunk();
```

Comment:
I am really not a fan of having to remap for everything, but I can imagine it is difficult to avoid here.